### ISSUE TITLE
Bundle - pipe broke by AntiVirus software

### ISSUE THREAD

**Issue by ipetrovic11** [wixtoolset/issues#5658]
I have come across one problem with Bundle installation and Antivirus Softwares.
[0D18:0660][2017-08-02T19:33:02]i011: Launched elevated engine process.
[0D18:0660][2017-08-02T19:33:02]e000: Error 0x800700e8: Failed to wait for child to connect to pipe.
[0D18:0660][2017-08-02T19:33:02]e000: Error 0x800700e8: Failed to connect to elevated child process.
[0D18:0660][2017-08-02T19:33:02]e000: Error 0x800700e8: Failed to actually elevate.
[0D18:0660][2017-08-02T19:33:02]e000: Error 0x800700e8: Failed to elevate.
[0D18:0660][2017-08-02T19:33:03]i399: Apply complete, result: 0x800700e8, restart: None, ba requested restart:  No

Displayed error is 0x800700e8 - The pipe is being closed.

So here is my explanation, when bundle starts it get scanned by AV in runtime and that is ok. Next step is that it extracts some other part of installation to TMP folder which is again EXE and this file also get scanned. These two processes now try to communicate but while scanning AV doesn't allow it and "main process" thinks that pipe is closed and shows this error.

FYI: all files are signed with valid Entrust EV signature. So that is not problem.

-----

I am think that there should another retry after first time so that this time antivirus is sure already sure that this is sage and process can be done.

Any thoughts about this?

**Comment by barnson** [comment]
Where we've seen this, A/V is terminating the elevated process.

**Comment by ipetrovic11** [comment]
A/V is scanning any file and probably creating some kind of sandbox where it is running the first time for first 15 seconds until they realize it is safe. What I am thinking is adding option that will keep trying it each 15 seconds for as many times as someone allows it.

I would be happy to take a look at this, can you point me in a code where this pipe initialization is?

I tried to repeat command that is elevating process "* -q -burn.elevated *" and while it is not done with scanning first time it gets scanned again and again. But when first time scan is done, this process starts normally without any interruption.

**Comment by ipetrovic11** [comment]
Ok, I can confirm that this is really working.
So I have created installation that on failure has install button again:
<img width="987" alt="screen shot 2017-08-02 at 9 07 20 pm" src="https://user-images.githubusercontent.com/3995534/28890429-640d519c-77c7-11e7-917d-86cd98b55ec9.png">
<img width="999" alt="screen shot 2017-08-02 at 9 07 37 pm" src="https://user-images.githubusercontent.com/3995534/28890432-67f16640-77c7-11e7-8e81-1f16b6546184.png">

So after error is shown I click Install button again and everything is installed.
So I am thinking about implementing something more smooth.

**Comment by barnson** [comment]
Interesting. The code involved is in the function `ElevationElevate` in https://github.com/wixtoolset/wix4/blob/develop/src/burn/engine/elevation.cpp.

**Comment by rseanhall** [comment]
Does the antivirus block the WiX installer as well?

I don't think it's practical for us to try to workaround antivirus, we just need to follow whatever process they have for false positives.

**Comment by ipetrovic11** [comment]
Yes same problem and you are solving this with try again as well.
<img width="922" alt="screen shot 2017-08-02 at 9 37 30 pm" src="https://user-images.githubusercontent.com/3995534/28891673-a35662fe-77cb-11e7-88de-0b233448c6d3.png">
<img width="922" alt="screen shot 2017-08-02 at 9 37 41 pm" src="https://user-images.githubusercontent.com/3995534/28891679-a646fd5c-77cb-11e7-8d81-374bc6624515.png">

So what I am thinking that you should just add one or two retries of function calling itself before ExitOnFailure(hr, "Failed to connect to elevated child process.");

Having a bit of trouble here compiling code but I will let you know as soon as I retry or if you could do that and send me for testing will be great.

**Comment by ipetrovic11** [comment]
I know that this is not clean code, but A/V software has created requirement for something like that unfortunately :(

**Comment by ipetrovic11** [comment]
Ok this definitely works for Avast

`// function definitions
int avCounter = 0;
extern "C" HRESULT ElevationElevate(
    __in BURN_ENGINE_STATE* pEngineState,
    __in_opt HWND hwndParent
    )
{
    Assert(BURN_MODE_ELEVATED != pEngineState->mode);
    Assert(!pEngineState->companionConnection.sczName);
    Assert(!pEngineState->companionConnection.sczSecret);
    Assert(!pEngineState->companionConnection.hProcess);
    Assert(!pEngineState->companionConnection.dwProcessId);
    Assert(INVALID_HANDLE_VALUE == pEngineState->companionConnection.hPipe);
    Assert(INVALID_HANDLE_VALUE == pEngineState->companionConnection.hCachePipe);

    HRESULT hr = S_OK;
    int nResult = IDOK;
    HANDLE hPipesCreatedEvent = INVALID_HANDLE_VALUE;

    nResult = pEngineState->userExperience.pUserExperience->OnElevate();
    hr = UserExperienceInterpretResult(&pEngineState->userExperience, MB_OKCANCEL, nResult);
    ExitOnRootFailure(hr, "UX aborted elevation requirement.");

    hr = PipeCreateNameAndSecret(&pEngineState->companionConnection.sczName, &pEngineState->companionConnection.sczSecret);
    ExitOnFailure(hr, "Failed to create pipe name and client token.");

    hr = PipeCreatePipes(&pEngineState->companionConnection, TRUE, &hPipesCreatedEvent);
    ExitOnFailure(hr, "Failed to create pipe and cache pipe.");

    LogId(REPORT_STANDARD, MSG_LAUNCH_ELEVATED_ENGINE_STARTING);

    do
    {
        nResult = IDOK;

        // Create the elevated process and if successful, wait for it to connect.
        hr = PipeLaunchChildProcess(pEngineState->sczBundleEngineWorkingPath, &pEngineState->companionConnection, TRUE, hwndParent);
        if (SUCCEEDED(hr))
        {
            LogId(REPORT_STANDARD, MSG_LAUNCH_ELEVATED_ENGINE_SUCCESS);

            hr = PipeWaitForChildConnect(&pEngineState->companionConnection);
            if(!SUCCEEDED(hr) && avCounter == 0){
                avCounter++;
                return ElevationElevate(pEngineState, hwndParent);;
            }
            ExitOnFailure(hr, "Failed to connect to elevated child process.");

            LogId(REPORT_STANDARD, MSG_CONNECT_TO_ELEVATED_ENGINE_SUCCESS);
        }
        else if (HRESULT_FROM_WIN32(ERROR_CANCELLED) == hr)
        {
            // The user clicked "Cancel" on the elevation prompt or the elevation prompt timed out, provide the notification with the option to retry.
            hr = HRESULT_FROM_WIN32(ERROR_INSTALL_USEREXIT);
            nResult = UserExperienceSendError(pEngineState->userExperience.pUserExperience, BOOTSTRAPPER_ERROR_TYPE_ELEVATE, NULL, hr, NULL, MB_ICONERROR | MB_RETRYCANCEL, IDNOACTION);
        }
    } while (IDRETRY == nResult);
    ExitOnFailure(hr, "Failed to elevate.");

LExit:
    ReleaseHandle(hPipesCreatedEvent);

    if (FAILED(hr))
    {
        PipeConnectionUninitialize(&pEngineState->companionConnection);
    }

    return hr;
}`

So first starts and wait for A/V to realize that file is actually safe.
Bundle still has open pipe and it starts process again so this time A/V don't need to test it since it is already marked as safe :) Here is log file so that you can see that it fails first time and second time it is connected:
[WorkPuls_20170803004638.txt](https://github.com/wixtoolset/issues/files/1195492/WorkPuls_20170803004638.txt)


There are probably few things that should also be ignore like creating pipe again but not sure I will let you take a look and let me know what you think.


I am in a bit of rush so because of our users so please help me build this right way for production.

**Comment by barnson** [comment]
Please make a pull request against https://github.com/wixtoolset/wix4 so it's easier to review.

**Comment by pablocar80** [comment]
Is there any workaround to this issue? Is it possible to create an installer that installs the .NET framework if needed, and works in computers with the most common antivirus programs?

**Comment by robmen** [comment]
@pablocar80 a solution was suggested above. Someone needs to put forth the effort to incorporate that change into a v4 PR (as @barnson points out above).

This does not seem to be a wide-spread problem with most common antivirus programs (especially when the Bundle is signed).

**Comment by pablocar80** [comment]
I can confirm that I have EV code signature from Digicert and running into this issue with Avast. It's very frustrating because users can't install and it's completely broken. Avast has 20% of the market according to https://www.statista.com/statistics/271048/market-share-held-by-antivirus-vendors-for-windows-systems/

**Comment by ipetrovic11** [comment]
Same happened to me, so that is why I proposed "hack" above.
I don't think it is the right solution but it works for now.

I am not so familiar with code and don't know your coding standards that is why I don't find my self appropriate to solve this directly in code base, but if there is someone interested I can describe why this solution is working, and he/she can implement it.

**Comment by pablocar80** [comment]
Thanks Ivan, where do we paste your code?

**Comment by ipetrovic11** [comment]
https://github.com/wixtoolset/wix4/blob/develop/src/burn/engine/elevation.cpp

**Comment by BMurri** [comment]
I simply took @ipetrovic11 changes to the wix3 code and applied them to the wix4 code.

**Comment by stukselbax** [comment]
We have faced the same problem (wix 3.11.0, 3.11.1) with [AVG](https://www.avg.com/), and added the retry logic to `OnApplyComplete` method, analyzing the error code and state of the Bootstrapper Application, like this:

```cpp
STDMETHODIMP_(int) OnApplyComplete(
    __in HRESULT hrStatus,
    __in BOOTSTRAPPER_APPLY_RESTART restart
)
{
    ...
    if (FAILED(hrStatus))
    {
        if (HRESULT_CODE(hrStatus) == ERROR_NO_DATA && !m_fStartedExecution)
        {
            BalLog(BOOTSTRAPPER_LOG_LEVEL_STANDARD, "The pipe is being closed error catched. Hope it is antivirus issue.");
            m_pEngine->Apply(m_pHwnd);
        }
    }
```

**Comment by stukselbax** [comment]
@BMurri As I can see the fix was done in https://github.com/wixtoolset/wix4 repo. Do you plan to merge same PR wixtoolset/wix4#258  to https://github.com/wixtoolset/wix3 ?

**Comment by BMurri** [comment]
@stukselbax: the enhancement is marked v4, and the pull request hasn't even been looked at. I only ported it from v3 to v4 in order to have @ipetrovic11's idea fully vetted/improved, as per the discussion above. I'm not claiming any ownership in his code, but I am interested in getting this issue addressed.

If this (or some variation thereof) is to be implemented in v3 is completely a question for triage to address, as they control what makes v3's bar (which, as it is nearing EOL, is quite high). Perhaps joining into their next meeting to argue your case would be a wonderful idea.

**Comment by BMurri** [comment]
@ipetrovic11 These are the meetings held by the owners of the WiX Toolset to triage incoming issues and discuss any other matters pressing to the toolset's developers (currently held approximately every other week, usually at 9am PST/PDT) on Skype and posted to the FireGiant blog and YouTube. I assume the next one would be on Aug 7th but @robmen would be a better person to ask about the scheduling. They are always announced on the wix-devs mailing list.

**Comment by aravindkumarmurugesan** [comment]
I am also facing the same issue. When use WIX Toolset 3.11 version while installing the setup face the above issue in avg anti virus installed machine. but If use WIX Toolset 3.9 version, Its working fine in the same machine.

May i know how to resolve this?

**Comment by stukselbax** [comment]
@aravindkumarmurugesan Have you tried to add retry logic, as [described earlier](https://github.com/wixtoolset/issues/issues/5658#issuecomment-407999659)?

**Comment by robmen** [comment]
I've put this issue in v4.0 release.

**Comment by emergencynap** [comment]
Seeing same with 3.11 - any plans to port it to 3.11 and release?  We are seeing this with Sophos, AVG and Avast.  Code is signed with EV as well.

**Comment by davidgemel** [comment]
After implementing the code from -> ipetrovic11 commented on Aug 2, 2017:

I ran into a problem where users when users did not have the .NET 4.6.1 as I had <PackageGroupRef Id="NetFx461Redist" />.  Instead of being prompted to install .NET 4.6.1 it was exiting with the error in the log but nothing in the UI:

> [0BA4:1138][2018-11-27T17:25:10]i000: Loading prerequisite bootstrapper application because managed host could not be loaded, error: 0x80131700.
> [0BA4:0BAC][2018-11-27T17:25:10]i000: Setting numeric variable 'WixStdBALanguageId' to value 1033
> [0BA4:0BAC][2018-11-27T17:25:10]i000: Setting version variable 'WixBundleFileVersion' to value '6.0.0.1001'
> [0BA4:0BAC][2018-11-27T17:25:10]e000: Error 0x8007057f: Failed to load theme controls.
> [0BA4:0BAC][2018-11-27T17:25:10]e000: Error 0x80004005: Failed to create main window.
> [0BA4:1138][2018-11-27T17:25:10]i500: Shutting down, exit code: 0x4005
> [0BA4:1138][2018-11-27T17:25:10]i000: The prerequisites were not successfully installed, error: 0x0. The bootstrapper application will be not reloaded.

Not sure if its related but I though I would put a note here.  I am now reversing the changes as it easier for me to explain an error code and tell them to turn off the virus scanner then to explain why the UI never came up.

**Comment by pablocar80** [comment]
This issue makes the install always fail anytime a user has an antivirus, even if the installer has an EV digital signature. I am missing something or is it not possible to use Bootstrapping functionality anymore? Is there any released version of Wix where it's possible to use a Bootstrapper?

**Comment by robmen** [comment]
@pablocar80 to be clear Burn operates correctly and operates correclty with most AV. At this point, it appears Avast AV is the only AV system that behaves in this particular way. Since it is unlikely that we'll be able to get Avast to change we'll need to update Burn to work around them.

However, this isn't a high priority for me personally (I have customers asking for other functionality and AFAIK you are not one of my customers) so until someone does all the work to incorporate a proper fix into WiX v4 and WiX v4 gets out there, this issue will wait until someone gets to it. I say this to be very clear: I personally think this issue is important that it'll need to get fixed in v4.0 (thus the milestone is set) so I will see it gets fixed. But, as noted above, it's not my current priority.

**Comment by emergencynap** [comment]
AVG suffers from the same issue by the way.  Building our own custom build of burn that is mentioned before in this thread does work for us, that’s what we’ve had to do.

**Comment by robmen** [comment]
@emergencynap that is a totally reasonable way to go at this time. Although, it'd be great if you've already cleaned up and verified the fix that you also set a pull request. Otherwise, someone else will need to repeat that amount of work you've already done.

**Comment by emergencynap** [comment]
I’ll ask someone on the team to get this done, didn’t work on it directly, just watching the issue.

**Comment by dchasman** [comment]
I know I am late to the party but we are also seeing this same issue with PC Matic SuperShield (a new one for me too) and I fully expect the workaround above to solve for that AV product too.

**Comment by rseanhall** [comment]
@davidgemel Can you please confirm that your bundle works properly on a computer that has no AV and doesn't have .NET 4.6.1? I'm hoping that there is a bug in your bundle since otherwise that would mean that even with the given solution there would still be other problems.

**Comment by davidgemel** [comment]
How I came across this scenario is we recently bumped our minimum requirements on our product from .NET 4.5.2 to 4.6.1. 

I can confirm I had a machine with .NET 4.5.2 installed, and no antivirus, and a custom burn using what ipetrovic11 commented on Aug 2, 2017.  When running that scenario I get the error in the comment above.  When I rolled back the changes I made to burn, the .NET installer dialog appears, and everything works properly.  As far as I can tell the those changes block my NetFx461Redist from working correctly. 

Note:
When users had .NET 4.6.1 already installed I had no issues, and anti virus fix worked.  My only issue is when the correct version of .NET is not installed.  My bootstrapper UI is .NET 4.6.1 and it makes sense that it won't appear unless the NetFx461Redist is called.

I put a copy of what my bundle looks like below (modified properties for the internet):

> <?xml version="1.0" encoding="UTF-8"?>
> <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:util="http://schemas.microsoft.com/wix/UtilExtension" xmlns:bal="http://schemas.microsoft.com/wix/BalExtension">
> 
>   <Bundle 
>     Name="MyStuff Deployment Center"
>     Version="$(var.InstallerVersion)" 
>     Manufacturer="My Company" 
>     UpgradeCode="$(var.UpgradeCode)" 
>     HelpUrl="http://www.mycompany.com"
>     AboutUrl="http://www.mycompany.com"
>     HelpTelephone="1-800-000-0000"
>     IconSourceFile="$(var.MyStuff.Controls.ProjectDir)Images\TopMenu\logon_logo.ico"
>     Condition="(VersionNT >= v6.1)"
>     >
>     
>     <BootstrapperApplicationRef Id="ManagedBootstrapperApplicationHost" >
>       <Payload SourceFile="$(var.BootstrapperUI.ProjectDir)BootstrapperCore.config"/>
>       <Payload SourceFile="$(var.BootstrapperUI.TargetDir)BootstrapperUI.dll"/>
>       <Payload SourceFile="$(var.MyStuff.Controls.TargetDir)MyStuff.Controls.dll"/>
>       <Payload SourceFile="$(var.MyStuff.Controls.TargetDir)MyStuff.Controls1.dll"/>
>       <Payload SourceFile="$(var.MyStuff.Controls.TargetDir)MyStuff.Controls2.dll"/>
>       <Payload SourceFile="$(var.MyStuff.Controls.TargetDir)MyStuff.Controls3.dll"/>
>       
>     </BootstrapperApplicationRef>
> 
>     <Variable Name="INSTALLDIR" bal:Overridable="yes" />
> 
>     <RelatedBundle Id="$(var.UpgradeCode)" Action="Detect"/>
> 		<Chain >
>       <PackageGroupRef Id="NetFx461Redist" />
>       <MsiPackage 
>         SourceFile="$(var.MyStuffSetup.TargetDir)MySetup.msi" 
>         Id="MyStuffPackageId"  
>         Compressed="yes" 
>         Cache="yes"
>         Permanent="no"
>          SuppressSignatureVerification="yes"
>          >
> 
>         <MsiProperty Name="TARGETDIR" Value="[INSTALLDIR]"   />
>         
> 
>       </MsiPackage>
> 		</Chain>
> 	</Bundle>
> </Wix>

**Comment by rseanhall** [comment]
@davidgemel Your issue is unrelated to this AV issue, your BootstrapperCore.config must not be properly configured. Please contact the wix-users mailing list for support.

**Comment by GoNzCiD** [comment]
Same problem here. I migrated my solution to WiX v4.0.0.5918 Release but problem is still present.
My solution is EV signed (before that it was blocked by Bitdefender too), AVG and Avast blocks bundle with this errors:
`[0598:0660][2018-12-18T11:09:28]i010: Launching elevated engine process.
[0598:0660][2018-12-18T11:09:58]i011: Launched elevated engine process.
[0598:0660][2018-12-18T11:09:58]e000: Error 0x800700e8: Failed to wait for child to connect to pipe.
[0598:0660][2018-12-18T11:09:58]e000: Error 0x800700e8: Failed to connect to elevated child process.
[0598:0660][2018-12-18T11:09:58]e000: Error 0x800700e8: Failed to actually elevate.
[0598:0660][2018-12-18T11:09:58]e000: Error 0x800700e8: Failed to elevate.
[0598:0660][2018-12-18T11:09:58]i000: Unsetting variable 'WixStdBARestartRequired'`

Machine is W10 x64 with net framework v4.7.1. I'm using HyperlinkTheme for bundle.

Any solution or workaround?

**Comment by rseanhall** [comment]
@dchasman I can't find a way to properly test with PC Matic Supershield without paying for it. From the limited testing I can do without a license, it appears to function differently than Avast/AVG as documented here. Please create a separate issue with whether you're signing your bundle, what the log file looks like, and also the behavior of our official [WiX v3.11.1 bundle](https://github.com/wixtoolset/wix3/releases/tag/wix3111rtm).

**Comment by GoNzCiD** [comment]
I've tested [this](https://github.com/wixtoolset/wix3/pull/477) solution with wix3.x and AVG still blocks the installation:
`[04E0:087C][2018-12-20T23:25:13]e000: Error 0x8007007e: WIXSTDBA: LoadBootstrapperBAFunctions() - Failed to load DLL C:\Users\admin\AppData\Local\Temp\{AE7F7037-EE0C-426E-A99D-8D980E8DE4BB}\.ba\bafunctions.dll
[04E0:11A0][2018-12-20T23:25:16]i010: Launching elevated engine process.
[04E0:11A0][2018-12-20T23:25:45]i011: Launched elevated engine process.
[04E0:11A0][2018-12-20T23:25:45]e000: Error 0x800700e8: Failed to wait for child to connect to pipe.
[04E0:11A0][2018-12-20T23:25:45]e000: Error 0x81f407d0: Failed to connect to elevated child process.
[0F80:1244][2018-12-20T23:26:04]e000: Error 0x80070643: Failed to install MSI package.
[0F80:1244][2018-12-20T23:26:04]e000: Error 0x80070643: Failed to execute MSI package.
[04E0:11A0][2018-12-20T23:26:04]e000: Error 0x80070643: Failed to configure per-machine MSI package.`

same solution migrated to wix4.x and with [this](https://github.com/wixtoolset/wix4/pull/262/files) modified files is working.

**Comment by rseanhall** [comment]
@GoNzCiD Do you know why there was 29 second gap between `Launching elevated engine process` and `Launched elevated engine process`? Did you just take that long to click OK for the UAC prompt? Is there a similar gap in time when you tested v4? The two pull requests are supposed to be identical, I wouldn't have expected Burn to change significantly enough in v4 for there to be a difference in behavior.

**Comment by GoNzCiD** [comment]
I'm testing it in a virtual machine and maybe it is not fast enought... I will try both solutions again and tell you results...

**Comment by LinearDynamics** [comment]
Also had to frig around with this for an hour before uninstalling Avast.

**Comment by jnm2** [comment]
@robmen 

> However, this isn't a high priority for me personally (I have customers asking for other functionality and AFAIK you are not one of my customers) so until someone does all the work to incorporate a proper fix into WiX v4 and WiX v4 gets out there, this issue will wait until someone gets to it. I say this to be very clear: I personally think this issue is important that it'll need to get fixed in v4.0 (thus the milestone is set) so I will see it gets fixed. But, as noted above, it's not my current priority.

When you get a chance, would you let us know what you are thinking with regard to merging the PR to fix this in v3 (v3 PR: https://github.com/wixtoolset/wix3/pull/477)? We're with the others in this thread; it's hit the three past releases of one of our products. I appreciate it!

**Comment by jnm2** [comment]
Avast, AVG, Sophos, and McAfee all crash WiX installers this way. It clearly can't be every WiX-based installer or a fix would be prioritized for WiX 3.11. Does anyone have a sense of what distinguishes the installers that crash from the ones that don't? E.g. we're bundling a .NET Framework web installer and a .NET Framework Windows Forms applications using DevExpress libraries.

**Comment by pablocar80** [comment]
@jnm2 the installers that crash are those that include the bootstrap tool Burn. There's no way to use it anymore as it won't run on computers with most antivirus. In my case, I was using bootstrap to install the NET Framework, and now instead I simply cancel the install with an error message that NET Framework is needed. This isn't as bad as I initially thought because Windows Update will always download the .NET Framework, at least on Windows 10.

**Comment by jnm2** [comment]
Does that mean (following the priorities mentioned above) that Rob's current customers aren't using Burn installers at all? If all Burn installers crash in the presence of all these mainstream antiviruses, surely that would necessitate a WiX 3.11 fix. Or is it only a subset of Burn installers- and if so, what distinguishes their contents from the Burn installers that don't crash?

**Comment by robmen** [comment]
@jnm2 almost every FireGiant customer uses Burn for their mainline product. None of them have encountered this problem. That is why it has not been a priority issue to fix from a FireGiant point of view. As we've never seen this issue repro, I can't say what distinguishes those products from the ones that have reportedly hit the issue. I expect this will eventually bubble up as an issue to tackle in our volunteer time at FireGiant but we have no customers reproducing it at this time.

**Comment by pablocar80** [comment]
Going back to @jnm2  question, my installer registers a Windows Service. Other folks running into the issue, are you doing that too?

**Comment by jnm2** [comment]
None of our installers with this issue contain Windows services.

**Comment by emergencynap** [comment]
This occurs if you are using an Extended Validation signing certificate, see my comment from October 2018.  Here is the EV certificates announcement https://blogs.msdn.microsoft.com/ie/2012/08/14/microsoft-smartscreen-extended-validation-ev-code-signing-certificates/

Once you sign your installer with that it fails in AV.  From memory what happens is this:
1.  Installer gets scanned by AV
2.  AV scans files but has not seen signature of this EV cert and does an internet based  lookup (read: slow).
3.  Wix times out due to AV scan and fails.

The solution is attached above somewhere but it involves compiling a custom version of Wix.  Basically retrying the operation or longer timeout winds up fixing the issue.

**Comment by jnm2** [comment]
We had the issue back when we didn't sign our installers or binaries. In some cases, the only thing we were bundling was the standard .NET Framework web installer. Would that have an EV cert, and if so, why doesn't everyone who bundles .NET run into this?

**Comment by xambuddy** [comment]
> generated

How did you managed to remove the previous installation in Programs and Features when you moved from wix3 to dotnetinstaller?

**Comment by xambuddy** [comment]
> Going back to @jnm2 question, my installer registers a Windows Service. Other folks running into the issue, are you doing that too?

How did you managed to remove the previous installation in Programs and Features when you moved from wix3 to dotnetinstaller?

**Comment by xambuddy** [comment]
> If anyone else runs into this issue -- the workaround I found was to keep the MSI generated by Wix, and then use a different bootstrapper from a different vendor (dotNetInstaller).

How did you managed to remove the previous installation in Programs and Features when you moved from wix3 to dotnetinstaller?

**Comment by ErwanDouaille** [comment]
Any news about this issue ?
Having this issue with a bundle installing 32 or 64 msi :/
Running Wix toolset v3.11.

**Comment by pablocar80** [comment]
@xambuddy I didn't have to deal with that problem because I found the issue before releasing.

**Comment by rseanhall** [comment]
Fix available in WiX v3.14.0.3910: https://wixtoolset.org/releases/v3-14-0-3910/. It's also fixed in the v4 micro repos.
